xds/src/main/java/io/grpc/xds/XdsLoadBalancer.java

```diff
@@ -145,30 +191,47 @@ public void handleResolvedAddresses(ResolvedAddresses resolvedAddresses) {
   private void handleNewConfig(XdsConfig xdsConfig) {
     String newBalancerName = xdsConfig.newBalancerName;
     LbConfig childPolicy = xdsConfig.childPolicy;
-    XdsComms xdsComms = null;
-    if (xdsLbState != null) { // may release and re-use/shutdown xdsComms from current xdsLbState
-      if (!newBalancerName.equals(xdsLbState.balancerName)) {
-        xdsComms = xdsLbState.shutdownAndReleaseXdsComms();
-        if (xdsComms != null) {
-          xdsComms.shutdownChannel();
-          xdsComms = null;
+    ManagedChannel lbChannel;
+    if (xdsLbState == null || !newBalancerName.equals(xdsLbState.balancerName)) {
```

Comment:
This logic is hard to read, what about declaring a private method `initLbChannel(newBalancerName)`, and then
```java
if (xdsLbState == null) {
  lbChannel = initLbChannel(newBalancerName);
} else if (!newBalancerName.equals(xdsLbState.balancerName)) {
  if (lrsWorking) {
    lrsClient.stopLoadReporting();
    lrsWorking = false;
  }
  ManagedChannel oldChannel = xdsLbState.shutdownAndReleaseChannel("Client shutdown");
  oldChannel.shutdown();
  lbChannel = initLbChannel(newBalancerName);
} else if (!Objects.equal(
    getPolicyNameOrNull(childPolicy),
    getPolicyNameOrNull(xdsLbState.childPolicy))) {
  lbChannel = xdsLbState.shutdownAndReleaseChannel("Changing load balancing mode");
} else { // effectively no change in policy, keep xdsLbState unchanged
  return;
}
```

Reply:
Done.